src/object_ref.jl

```diff
@@ -35,13 +38,58 @@ function has_owner(obj_ref::ObjectRef)
     return !isempty(ray_jll.SerializeAsString(owner_address))
 end
 
+function _register_ownership(obj_ref::ObjectRef, outer_obj_ref::Union{ObjectRef,Nothing})
+    worker = ray_jll.GetCoreWorker()
+
+    outer_object_id = if outer_obj_ref !== nothing
+        outer_obj_ref.oid
+    else
+        ray_jll.Nil(ray_jll.ObjectID)
+    end
+
+    if !isnothing(obj_ref.owner_address) && !has_owner(obj_ref)
+        # https://github.com/ray-project/ray/blob/ray-2.5.1/python/ray/_raylet.pyx#L3329
+        # https://github.com/ray-project/ray/blob/ray-2.5.1/src/ray/core_worker/core_worker.h#L543
+        ray_jll.RegisterOwnershipInfoAndResolveFuture(worker, obj_ref.oid, outer_object_id,
+                                                      obj_ref.owner_address,
+                                                      obj_ref.serialized_object_status)
+    end
+
+    return nothing
+end
+
 # We cannot serialize pointers between processes
 function Serialization.serialize(s::AbstractSerializer, obj_ref::ObjectRef)
+    worker = ray_jll.GetCoreWorker()
+
+    hex_str = hex_identifier(obj_ref)
+    owner_address = ray_jll.Address()
+    serialized_object_status = StdString()
+
+    # Prefer serializing ownership information from the core worker backend
+    ray_jll.GetOwnershipInfo(worker, obj_ref.oid, CxxPtr(owner_address), CxxPtr(serialized_object_status))
+    owner_address_str = String(ray_jll.SerializeAsString(owner_address))
+    serialized_object_status = String(serialized_object_status)
```

Comment:
kinda feels like this should be one function in the jll...

Reply:
My opinion on this is that the JLL should just mostly just contain a direct C++ to Julia interface and have Ray.jl performing most of the calls. With this approach the internal code for Ray.jl can mirror the Python Ray code quite closely. There are some reasons to push more things onto the C++ side for performance reasons.